bcl2fastq/lib/bcl2fastq_utils.py

```diff
@@ -134,32 +140,49 @@ def get_bases_mask_per_lane_from_samplesheet(samplesheet, index_lengths):
         :param samplesheet: samplesheet to fetch the index lengths from
         :param index_lengths: dict of index lengths (e.g. "{1: 7, 2: 8}"),
         normally parsed from run meta data.
+        :param is_single_read: True if this is a single read run, else false.
         :return a dict of the lane and base mask to use, e.g.:
-                 { 1:"y*,iiiiiiiin*,iiiiiiiin*,y*" , 2:"y*,iiiiii,n*,y*  [etc] }
+                 { 1:"y*,i7n*,i7n*,y*" , 2:"y*,i5,n*,y*  [etc] }
         """
 
-        def pad_with_ignore(length_of_index_in_samplesheet, length_of_index_read):
+        def build_index_string(length_tuple):
+            """
+            Builds the index mask string
+            :param length_tuple: a tuple of the length of the index in the samplesheet and in the read, e.g. (3, 5)
+            :return: a index string, e.g. "i5n*" or "n*" or "i3", depending on the situation.
+            """
+            length_of_index_in_samplesheet = length_tuple[0]
+            length_of_index_read = length_tuple[1]
             difference = length_of_index_read - length_of_index_in_samplesheet
+
             assert difference >= 0, "Sample sheet indicates that index is longer than what was read by the sequencer!"
```

Comment:
I believe `assert` statements is not considered stable for production code: ```Assertions should *not* be used to test for failure cases that can occur because of bad user input or operating system/environment failures, such as a file not being found. Instead, you should raise an exception, or print an error message, or whatever is appropriate. One important reason why assertions should only be used for self-tests of the program is that assertions can be disabled at compile time.``` (from https://wiki.python.org/moin/UsingAssertionsEffectively)